lighthouse-core/gather/gatherers/mixed-content.js

```diff
@@ -47,58 +50,70 @@ class MixedContent extends Gatherer {
+          !this.ids.has(event.interceptionId)) {
+        this.ids.add(event.interceptionId);
+        event.request.url = this.upgradeURL(event.request.url);
+        driver.sendCommand('Network.continueInterceptedRequest', {
+          interceptionId: event.interceptionId,
+          rawResponse: Buffer.from(
+              `HTTP/1.1 302 Found\r\nLocation: ${event.request.url}\r\n\r\n`,
+              'utf8').toString('base64'),
+        });
+      } else {
+        driver.sendCommand('Network.continueInterceptedRequest', {
+          interceptionId: event.interceptionId,
+        });
+      }
+    };
+
+    return onRequestIntercepted;
   }
 
-  beforePass(options) {
-    const driver = options.driver;
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   */
+  async beforePass(passContext) {
+    const driver = passContext.driver;
 
     // Attempt to load the HTTP version of the page.
     // The base URL can be brittle under redirects of the page, but the worst
     // case is the gatherer accidentally upgrades the final base URL to
     // HTTPS and loses the ability to check upgrading active mixed content.
-    options.url = this.downgradeURL(options.url);
-    this.url = options.url;
+    passContext.url = this.downgradeURL(passContext.url);
+    this.url = passContext.url;
+    this._onRequestIntercepted = this._getRequestInterceptor(this.url, driver);
 
-    driver.sendCommand('Network.enable', {});
-    driver.on('Network.requestIntercepted',
-      this._onRequestIntercepted.bind(this, driver));
-    driver.sendCommand('Network.setCacheDisabled', {cacheDisabled: true});
-    driver.sendCommand('Network.setRequestInterception',
-      {patterns: [{urlPattern: '*'}]});
+    await driver.sendCommand('Network.enable');
+    driver.on('Network.requestIntercepted', this._onRequestIntercepted);
+    await driver.sendCommand('Network.setCacheDisabled', {cacheDisabled: true});
+    await driver.sendCommand('Network.setRequestInterception', {patterns: [{urlPattern: '*'}]});
   }
 
-  afterPass(options, _) {
-    const driver = options.driver;
-    return Promise.resolve().then(_ => driver.sendCommand(
-        'Network.setRequestInterception', {patterns: []})
-    ).then(_ => driver.off(
-        'Network.requestIntercepted',
-        this._onRequestIntercepted.bind(this, driver))
-    ).then(driver.sendCommand(
-        'Network.setCacheDisabled', {cacheDisabled: false})
-    ).then(_ => {
-      return {url: options.url};
-    });
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @return {Promise<{url: string}>}
+   */
+  async afterPass(passContext) {
+    const driver = passContext.driver;
+    await driver.sendCommand('Network.setRequestInterception', {patterns: []});
+    if (this._onRequestIntercepted) {
+      driver.off('Network.requestIntercepted', this._onRequestIntercepted);
+    }
+    await driver.sendCommand('Network.setCacheDisabled', {cacheDisabled: false});
```

Comment:
yeah, technically it was running, just messing up the promise chain :) In theory some kind of racy thing could happen (or if there was an error, you'd end up with an unhandled promise rejection), but since we don't run this in smoke testing we'll never know! :)